v3/src/components/data-display/components/legend/legend.tsx

```diff
@@ -34,7 +35,13 @@ export const Legend = function Legend({
             : attrType === 'numeric' || attrType === 'date'
               ? <NumericLegend
                   layerIndex={layerIndex}
-                  setDesiredExtent={setDesiredExtent}/> : null
+                  setDesiredExtent={setDesiredExtent}/>
+              : attrType === 'color'
+                ? <ColorLegend
+                    layerIndex={layerIndex}
+                    setDesiredExtent={setDesiredExtent}
+                  />
+                : null
```

Comment:
Given the complexity of this nested ternary, something like the following would be an improvement:
```typescript
    legendComponentMap: Partial<Record<string, React.ComponentType<IBaseLegendProps>>> = {
      categorical: CategoricalLegend,
      color: ColorLegend,
      date: NumericLegend,
      numeric: NumericLegend
    },
    LegendComponent = attrType && legendComponentMap[attrType]
``` 
up above and then this becomes:
```typescript
        <LegendComponent layerIndex={layerIndex} setDesiredExtent={setDesiredExtent}/>
```